Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -173,21 +178,74 @@ static CFStringRef _CFStringFromLocalizedString(IDWriteLocalizedStrings* localiz
+    DWRITE_FONT_STYLE style = DWRITE_FONT_STYLE_NORMAL;
+    DWRITE_FONT_STRETCH stretch = DWRITE_FONT_STRETCH_NORMAL;
     if (font != nil) {
-        fontSize = [font pointSize];
+        fontSize = CTFontGetSize(font);
+        __InitDWriteFontPropertiesFromName(CTFontCopyName(font, kCTFontFullNameKey), &weight, &stretch, &style);
+        CFStringRef name = _DWriteGetFamilyNameForFontName(CTFontCopyName(font, kCTFontFullNameKey));
+        fontName.resize(CFStringGetLength(name) + 1);
+        CFStringGetCharacters(name, CFRangeMake(0, fontName.size()), reinterpret_cast<UniChar*>(fontName.data()));
     }
 
     ComPtr<IDWriteTextFormat> textFormat;
-    THROW_IF_FAILED(dwriteFactory->CreateTextFormat(c_defaultFontName,
-                                                    NULL,
-                                                    DWRITE_FONT_WEIGHT_REGULAR,
-                                                    DWRITE_FONT_STYLE_NORMAL,
-                                                    DWRITE_FONT_STRETCH_NORMAL,
-                                                    fontSize,
-                                                    c_defaultUserLanguage,
-                                                    &textFormat));
+    THROW_IF_FAILED(
+        dwriteFactory->CreateTextFormat(fontName.data(), NULL, weight, style, stretch, fontSize, c_defaultUserLanguage, &textFormat));
+
+    CTParagraphStyleRef settings =
+        static_cast<CTParagraphStyleRef>([attribs valueForKey:static_cast<NSString*>(kCTParagraphStyleAttributeName)]);
+    if (settings) {
+        CTTextAlignment alignment = kCTNaturalTextAlignment;
+        if (CTParagraphStyleGetValueForSpecifier(settings, kCTParagraphStyleSpecifierAlignment, sizeof(CTTextAlignment), &alignment)) {
+            DWRITE_TEXT_ALIGNMENT dwriteAlignment = DWRITE_TEXT_ALIGNMENT_LEADING;
+            switch (alignment) {
+                case kCTRightTextAlignment:
+                    dwriteAlignment = DWRITE_TEXT_ALIGNMENT_TRAILING;
+                    break;
+                case kCTCenterTextAlignment:
+                    dwriteAlignment = DWRITE_TEXT_ALIGNMENT_CENTER;
+                    break;
+                case kCTJustifiedTextAlignment:
+                    dwriteAlignment = DWRITE_TEXT_ALIGNMENT_JUSTIFIED;
+                    break;
+                case kCTLeftTextAlignment:
+                case kCTNaturalTextAlignment:
+                default:
+                    dwriteAlignment = DWRITE_TEXT_ALIGNMENT_LEADING;
+            }
+
+            THROW_IF_FAILED(textFormat->SetTextAlignment(dwriteAlignment));
+        }
+
+        CTWritingDirection direction;
+        if (CTParagraphStyleGetValueForSpecifier(settings, kCTParagraphStyleSpecifierBaseWritingDirection, sizeof(direction), &direction)) {
+            DWRITE_READING_DIRECTION dwriteDirection = DWRITE_READING_DIRECTION_LEFT_TO_RIGHT;
+            switch (direction) {
+                case kCTWritingDirectionRightToLeft:
```

Comment:
we need to add a test in the test app for the RTL text, given we are adding this now.